v2/apiserver/internal/core/jobs.go

```diff
@@ -253,15 +270,24 @@ func (j *jobsService) Create(
 	if err != nil {
 		return errors.Wrapf(err, "error retrieving event %q from store", eventID)
 	}
-	if _, ok := event.Worker.Job(job.Name); ok {
-		return &meta.ErrConflict{
-			Type: JobKind,
-			ID:   job.Name,
-			Reason: fmt.Sprintf(
-				"Event %q already has a job named %q.",
-				eventID,
-				job.Name,
-			),
+	if originalJob, ok := event.Worker.Job(job.Name); ok {
+		// If we have a match and this isn't a retry, return an error
+		if event.Labels == nil || event.Labels[RetryLabelKey] == "" {
+			return &meta.ErrConflict{
+				Type: JobKind,
+				ID:   job.Name,
+				Reason: fmt.Sprintf(
+					"Event %q already has a job named %q.",
+					eventID,
+					job.Name,
+				),
+			}
+		}
```

Comment:
Are we still missing a check somewhere in this vicinity?

I think even if the event itself is a try, creating a job with the same name as an existing one should raise a conflict if the job that would be replaced wasn't inherited.

Reply:
Oh, in that case, I was thinking we'd just proceed with re-scheduling the job with the same name, instead of returning a conflict error.  WDYT?

Reply:
But wouldn't that let a single worker create a job with the same name more than once as long as the even is a retry?

Reply:
Ahh yes, thank you.  Took me a bit to grok.  Will update.

Reply:
I've changed it back to just returning the ErrConflict for all cases except for when the job is a retry and its configuration matches the original.  I'd rather revisit the corner cases around this at a future date when needed.

Reply:
sgtm

Reply:
But the logic looks wrong. Shouldn't it be this this (difference is `||` vs `&&`):

```
if event.Labels != nil || event.Labels[RetryLabelKey] != "" {
```

Reply:
Also, I do think we can do something more here in a follow-up.

Reply:
Good catch, it should definitely be `&&` --  a mistake carried over when I had this setup to check the reverse.